vortex-buffer/src/buffer_mut.rs

```diff
@@ -55,6 +55,7 @@ impl<T> BufferMut<T> {
     pub fn zeroed_aligned(len: usize, alignment: Alignment) -> Self {
         let mut bytes = BytesMut::zeroed((len * size_of::<T>()) + *alignment);
         bytes.advance(bytes.as_ptr().align_offset(*alignment));
+        unsafe { bytes.set_len(len * size_of::<T>()) };
```

Comment:
why was this added?

Reply:
I thought it was fixed here, surprised it still shows in the diff: https://github.com/spiraldb/vortex/pull/2099/files

Reply:
I pulled this in